src/simplify_dyn_ops.cpp

```diff
@@ -26,10 +26,131 @@
 #include <migraphx/matcher.hpp>
 #include <migraphx/make_op.hpp>
 #include <migraphx/literal.hpp>
+#include <migraphx/op/resize.hpp>
+
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdocumentation-unknown-command"
+/**
+ * Convert a Resize op. with Nearest mode to an implementation using Gather op.
+ * From:  resize[scales={...}/sizes={...},](static, constant)
+ * To:  
+ * @0 = @literal{ ... } computed_indices
+ * ... 
+ * @2 = reshape[dims={45}](X) 1-dimensional
+ * @3 = gather[axis=0](@2,@0) 
+ * 
+ * At the time of writing, this conversion is required for GPU targets because there
+ * is not direct a GPU implementation of the Resize operation.
+ * This matcher depends on a split_single_dyn_dim pass being run before it, which
+ * will convert any dynamic-batch input to static inputs and make this conversion possible.
+ * 
+ *   At time of writing, Resize allows either 1 or 2 inputs
+ * but the 1-input case is never created by Onnx parsing.
+ */
+#pragma("GCC diagnostic pop")
+
+struct find_resize_static
+{
+ 
+    auto matcher() const
+    {
+        std::cout << "calling resize matcher...........................\n";
+
+        // Will this work with either dynamic or static input 0?
+        return match::name("resize")(match::nargs(2), 
+                match::arg(0)(match::static_shape()),
+                match::arg(1)(match::is_constant()));  
+
+        // How to add the other case:  only 1 input arg
```

Comment:
You can use the `any_of` matcher to list out the different case.

Reply:
Comment removed because at this point, we don't need to match the 1-argument case in operational use.  The Onnx parser doesn't make a Resize op. when it only gets 1 argument, and Resize ops with 1 argument can only be created programmatically (i.e. by a unit test).